### ISSUE TITLE
Provide full path to build directory via a variable

### ISSUE THREAD

**Issue by huguesalary** [concourse/concourse#4281]
## What challenge are you facing?

TL;DR: I need my environment variable `GOOGLE_APPLICATION_CREDENTIALS` (defined in `params`) to point to the **absolute path** of my build directory.

I have the following task:

```
platform: linux

image_resource:
  type: docker-image
  source:
    repository: gcr.io/kaniko-project/executor:latest

inputs:
- name: source
- name: credentials

outputs:
- name: image

run:
  path: /kaniko/executor
  args:
  - --dockerfile=source/Dockerfile
  - --context=source/
  - --tarPath=image/image.tar
  - --no-push
params:
  GOOGLE_APPLICATION_CREDENTIALS: credentials/google_application_credentials.json
```

- The `credentials` input is a folder that contains
- a `google_application_credentials.json` file, 
- which is used by `/kaniko/executor`, 
- via the environment variable `GOOGLE_APPLICATION_CREDENTIALS`, to authenticate against a private docker registry.

Unfortunately, when this task run, the environment variable value is a **relative** path (`credentials/google_application_credentials.json`) to the *actual* **build directory** in which the task is running (in this case `/tmp/build/sha1/`).

That is, at the time of execution the `/kaniko/executor` binary will try to read the file at `credentials/google_application_credentials.json`, which isn't a valid path since this environment variable is supposed to point to an absolute path.

I need my environment variable `GOOGLE_APPLICATION_CREDENTIALS` (defined in `params`) to point to the **absolute path** of my build directory.

The common pattern in the concourse eco-system is to write a thin shell wrapper around the executable provided by the docker image as such:

```
[OMMITED CONFIGURATION FOR EASE OF READ]

run:
  # Run a shell instead of the binary directly
  path: /bin/sh 
  args:
  - -c
  - |
    set -e
    readonly WORKSPACE_DIR="$(pwd)"
    export readonly GOOGLE_APPLICATION_CREDENTIALS="${WORKSPACE_DIR}/credentials/google_application_credentials.json"
    exec /kaniko/executor \
      --dockerfile=source/Dockerfile \
      --context=source/ \
      --tarPath=image/image.tar \
      --no-push
```

However, this solution **only works if the image contains a shell**.

In this instance, the `gcr.io/kaniko-project/executor:latest` does not contain a shell, and this solution does not apply.

## What would make this better?

Concourse-CI needs to provide a way for pipelines to be fed their build directory. My limited knowledge about concourse does not allow me to recommend one way to do it over another, however, as a concourse user, here's what would feel the most natural: a concourse provided variable like `((concourse-build-folder))`

```
[OMMITED CONFIGURATION FOR EASE OF READ]

run:
  path: /kaniko/executor
  args:
  - --dockerfile=source/Dockerfile
  - --context=source/
  - --tarPath=image/image.tar
  - --no-push
params:
  GOOGLE_APPLICATION_CREDENTIALS: ((concourse-build-folder))/credentials/google_application_credentials.json
```

A variable provided by concourse would be in line with the philosophy of preventing users from tying themselves to how concourse runs builds, yet, give them the possibility of discovering the `build path` at runtime, without having to resort to anti-pattern solutions.

## Are you interested in implementing this yourself?

I would be more than happy to help implementing any solution allowing me to write my pipelines without expecting the presence of a shell.

**Comment by vito** [comment]
Maybe we could allow specifying an absolute path from which to run?

We currently allow you to specify a path in [`run.dir`](https://concourse-ci.org/tasks.html#task-run-dir) like so:

```yaml
run:
  path: /kaniko/executor
  dir: /kaniko
  args: # ...
```

...however, right now the `dir` value gets joined with the configured workdir (`/tmp/build/blahblah`), and absolute paths are just not respected at all.

I think it would be possible to just respect absolute paths and mount the inputs under that path instead.

**Comment by huguesalary** [comment]
That sounds reasonable to me. I'm trying to think of reasons this wouldn't be a good solution but can't think of any right now.

**Comment by vito** [comment]
gave feedback on https://github.com/concourse/concourse/pull/4299 - hope it helps!

**Comment by huguesalary** [comment]
Ha thanks much! I'll dust of my concourse dev environment and see if I can implement this.

**Comment by fran6co** [comment]
Wouldn't be useful to still provide the absolute path to the working directory? For example when adding caching support it would be useful to be able to do something like this:

```
          params:
              PIP_CACHE_DIR: ((working_dir))/.pip-cache
          caches:
              - path: .pip-cache
```

A lot of tools allow to set the cache path via environment variables but the directory where the package managers are executed from can vary a lot

**Comment by deephack1982** [comment]
I'm just bumping my head against this exact issue with setting GOOGLE_APPLICATION_CREDENTIALS. I have a relative path which works right now but it feels a bit fragile and could break from task to task in the one job.
